MonkeyRunner tries to location layer

We should add `!= null` in onStop:

```
// CRASH: com.mapbox.mapboxsdk.plugins.testapp (pid 27898)
// Short Msg: java.lang.NullPointerException
// Long Msg: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.mapbox.mapboxsdk.plugins.locationlayer.LocationLayerPlugin.onStop()' on a null object reference
// Build Label: google/marlin/marlin:8.0.0/OPR3.170623.008/4294783:user/release-keys
// Build Changelist: 4294783
// Build Time: 1503515582000
// java.lang.RuntimeException: Unable to stop activity {com.mapbox.mapboxsdk.plugins.testapp/com.mapbox.mapboxsdk.plugins.testapp.activity.location.CompassListenerActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.mapbox.mapboxsdk.plugins.locationlayer.LocationLayerPlugin.onStop()' on a null object reference
//     at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4400)
//     at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4456)
//     at android.app.ActivityThread.-wrap5(Unknown Source:0)
//     at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1654)
//     at android.os.Handler.dispatchMessage(Handler.java:105)
//     at android.os.Looper.loop(Looper.java:164)
//     at android.app.ActivityThread.main(ActivityThread.java:6541)
//     at java.lang.reflect.Method.invoke(Native Method)
//     at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
//     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
// Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.mapbox.mapboxsdk.plugins.locationlayer.LocationLayerPlugin.onStop()' on a null object reference
//     at com.mapbox.mapboxsdk.plugins.testapp.activity.location.CompassListenerActivity.onStop(CompassListenerActivity.java:77)
//     at android.app.Instrumentation.callActivityOnStop(Instrumentation.java:1374)
//     at android.app.Activity.performStop(Activity.java:7150)
//     at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4395)
//     ... 9 more
// 
```

--- a/app/src/main/java/com/mapbox/mapboxsdk/plugins/testapp/activity/location/CompassListenerActivity.java
+++ b/app/src/main/java/com/mapbox/mapboxsdk/plugins/testapp/activity/location/CompassListenerActivity.java
@@ -74,7 +74,9 @@ public class CompassListenerActivity extends AppCompatActivity implements OnMapR
   protected void onStop() {
     super.onStop();
     mapView.onStop();
-    locationLayerPlugin.onStop();
+    if (locationLayerPlugin != null) {
+      locationLayerPlugin.onStop();
+    }
   }
 
   @Override

